internal/services/config/config_test.go

```diff
@@ -0,0 +1,266 @@
+
+	"agola.io/agola/internal/util"
+)
+
+func TestParseConfig(t *testing.T) {
+	tests := []struct {
+		name     string
+		services []string
+		in       string
+		err      error
+	}{
+		{
+			name: "test all-base and executor",
+			services: []string{"all-base", "executor"},
+			in: `
+gateway:
+  apiExposedURL: "http://localhost:8000"
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  gitserverURL: "http://localhost:4003"
+
+  web:
+    listenAddress: ":8000"
+  tokenSigning:
+    method: hmac
+    key: supersecretsigningkey
+  adminToken: "admintoken"
+
+scheduler:
+  runserviceURL: "http://localhost:4000"
+
+notification:
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  etcd:
+    endpoints: "http://localhost:2379"
+
+configstore:
+  dataDir: /data/agola/configstore
+  etcd:
+    endpoints: "http://localhost:2379"
+  objectStorage:
+    type: posix
+    path: /agola/configstore/ost
+  web:
+    listenAddress: ":4002"
+
+runservice:
+  #debug: true
+  dataDir: /opt/data/agola/runservice
+  etcd:
+    endpoints: "http://localhost:2379"
+  objectStorage:
+    type: posix
+    path: /agola/runservice/ost
+  web:
+    listenAddress: ":4000"
+
+executor:
+  allowPrivilegedContainers: true
+  dataDir: /data/agola/executor
+  toolboxPath: ./bin
+  runserviceURL: "http://localhost:4000"
+  web:
+    listenAddress: ":4001"
+  activeTasksLimit: 5
+  driver:
+    type: docker
+
+gitserver:
+  dataDir: /data/agola/gitserver
+  gatewayURL: "http://localhost:8000"
+  web:
+    listenAddress: ":4003"`,
+		},
+		{
+			name: "test all-base without executor",
+			services: []string{"all-base"},
+			in: `
+gateway:
+  apiExposedURL: "http://localhost:8000"
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  gitserverURL: "http://localhost:4003"
+
+  web:
+    listenAddress: ":8000"
+  tokenSigning:
+    method: hmac
+    key: supersecretsigningkey
+  adminToken: "admintoken"
+
+scheduler:
+  runserviceURL: "http://localhost:4000"
+
+notification:
+  webExposedURL: "http://localhost:8000"
+  runserviceURL: "http://localhost:4000"
+  configstoreURL: "http://localhost:4002"
+  etcd:
+    endpoints: "http://localhost:2379"
+
+configstore:
+  dataDir: /data/agola/configstore
+  etcd:
+    endpoints: "http://localhost:2379"
+  objectStorage:
+    type: posix
+    path: /agola/configstore/ost
+  web:
+    listenAddress: ":4002"
+
+runservice:
+  dataDir: /data/agola/runservice
+  etcd:
+    endpoints: "http://localhost:2379"
+  objectStorage:
+    type: posix
+    path: /agola/runservice/ost
+  web:
+    listenAddress: ":4000"
+
+gitserver:
+  dataDir: /data/agola/gitserver
+  gatewayURL: "http://localhost:8000"
+  web:
+    listenAddress: ":4003"`,
+		},
+		{
+			name: "test gateway, scheduler and notification",
```

Comment:
I'll reword it to something like: `test config for gateway, scheduler and notification`